src/shared/port-provider.js

```diff
@@ -151,29 +116,36 @@ export class PortProvider {
     const handleRequest = event => {
       const { data, origin, source } = /** @type {MessageEvent} */ (event);
 
-      if (!isSourceWindow(source)) {
+      // Check that message is a port request intended for us.
+      if (!isMessage(data) || data.type !== 'request') {
         return;
       }
 
-      const match = this._allowedMessages.find(
-        ({ allowedOrigin, ...allowedMessage }) =>
-          this._messageMatches({
-            allowedMessage,
-            allowedOrigin,
-            data,
-            origin,
-          })
-      );
-
-      if (match === undefined) {
-        return;
+      // Validate request details and source.
+      const { frame1, frame2 } = data;
+      const channel = /** @type {Channel} */ (`${frame1}-${frame2}`);
+      if (!allowedChannels.includes(channel)) {
+        throw new Error(`Port request has unsupported channel ${channel}`);
       }
 
-      const { frame1, frame2 } = match;
-      const channel = /** @type {Channel} */ (`${frame1}-${frame2}`);
+      if (!isSourceWindow(source)) {
+        throw new Error(
+          `Port request for ${channel} came from a non-Window source`
+        );
+      }
+
+      const isAllowedOrigin =
+        // Guest and host frames may be hosted on any origin.
+        ['guest', 'host'].includes(frame1) ||
+        // The sidebar and notebook apps must be loaded from the expected origin.
+        origin === this._hypothesisAppsOrigin;
+      if (!isAllowedOrigin) {
+        throw new Error(
```

Comment:
I don't agree with this approach. We are adding errors just to debug the current situation. The problem that I see with this is that the errors are not going to be removed from the code. I think the correct approach is to use `sendError` which is clearly a debugging statement, that can perfectly be removed from the code in the future and doesn't need to be tested.